### ISSUE TITLE
Non-encodables are silently ignored

### ISSUE THREAD

**Issue by seprich** [py-bson/bson#36]
```python
In [20]: def foo():
   ....:     print('testing')
   ....:     

In [21]: bson.dumps({'test': foo})
Out[21]: b'\x05\x00\x00\x00\x00'
```
As the above example shows, the dumps function silently ignores content which is not encodable. This behaviour is somewhat unexpected, since e.g. json.dumps does raise a TypeError in this situation. Perhaps provide an option to set silent filtering but by default raise errors.

**Comment by seprich** [comment]
so ok... in dev branch you have this fixed already :) nice. Till the next release then.

**Comment by Parkayun** [comment]
I will release that soon :)
thx.

**Comment by Parkayun** [comment]
@seprich I released 0.4.2.
